# 02 Trend Across Time

## import modules and dataset

In [1]:
import pandas as pd
import altair as alt
# alt.renderers.enable('jupyterlab')

In [2]:
trials_clean = pd.read_csv('./export/trials_clean.csv')
trials_clean.sample(20)

year  decade  century  tried  deaths                       city  \
1704   1711.0    1710     1700      1     1.0                  Straubing   
10338  1645.0    1640     1600      1     NaN                        NaN   
1154   1629.0    1620     1600      1     NaN                 Zoschingen   
1062   1619.0    1610     1600      3     NaN      Neustadt an der Donau   
7671   1670.0    1670     1600      1     NaN  Neuhausen auf den Fildern   
3442   1629.0    1620     1600      1     0.0                        NaN   
5266   1662.0    1660     1600      1     0.0                        NaN   
10305  1630.0    1630     1600      1     NaN                        NaN   
10107  1649.0    1640     1600      1     NaN                        NaN   
2294   1615.0    1610     1600      1     0.0                      Namur   
1322   1665.0    1660     1600      2     2.0                   Lengdorf   
6881   1557.0    1550     1500      1     1.0        Zell am Harmersbach   
5557   1661.0    1660     1600      1     0.0                        NaN   
5719   1678.0    1670     1600      1     0.0                        NaN   
4615   1650.0    1650     1600      1     NaN                        NaN   
9303   1634.0    1630     1600      1     NaN                    Grundau   
7369   1627.0    1620     1600      4     4.0               Ortenaukreis   
374       NaN    1610     1600      2     2.0                        NaN   
9498   1586.0    1580     1500      1     1.0                        NaN   
5821   1695.0    1690     1600      1     NaN                        NaN   

                 gadm.adm2          gadm.adm1       gadm.adm0       lon  \
1704         Mittelfranken             Bayern         Germany  12.58020   
10338                Essex            England  United Kingdom       NaN   
1154         Mittelfranken             Bayern         Germany  10.32550   
1062         Mittelfranken             Bayern         Germany  11.76710   
7671             Stuttgart  Baden-Wurttemberg         Germany   9.27583   
3442   West Dunbartonshire           Scotland  United Kingdom       NaN   
5266          Renfrewshire           Scotland  United Kingdom       NaN   
10305        Hertfordshire            England  United Kingdom       NaN   
10107                  NaN    Vastra Gotaland          Sweden       NaN   
2294                 Namur           Wallonie         Belgium   4.87199   
1322         Mittelfranken             Bayern         Germany  12.04730   
6881              Freiburg  Baden-Wurttemberg         Germany   8.06565   
5557      Scottish Borders           Scotland  United Kingdom       NaN   
5719             Edinburgh           Scotland  United Kingdom       NaN   
4615     South Lanarkshire           Scotland  United Kingdom       NaN   
9303             Darmstadt             Hessen         Germany   9.11065   
7369              Freiburg  Baden-Wurttemberg         Germany   7.94714   
374                    NaN       Sankt Gallen     Switzerland       NaN   
9498                  Kent            England  United Kingdom       NaN   
5821              Highland           Scotland  United Kingdom       NaN   

           lat          record.source  
1704   48.8777       Behringer (1987)  
10338      NaN        Valletta (2000)  
1154   48.6736       Behringer (1987)  
1062   48.8030       Behringer (1987)  
7671   48.6821       Midelfort (1972)  
3442       NaN  Goodare et al. (2003)  
5266       NaN  Goodare et al. (2003)  
10305      NaN        Valletta (2000)  
10107      NaN          Sörlin (1999)  
2294   50.4674        Brouette (1953)  
1322   48.2578       Behringer (1987)  
6881   48.3472       Midelfort (1972)  
5557       NaN  Goodare et al. (2003)  
5719       NaN  Goodare et al. (2003)  
4615       NaN  Goodare et al. (2003)  
9303   50.2023           Niess (1982)  
7369   48.3645       Midelfort (1972)  
374        NaN           Bader (1945)  
9498       NaN       Notestein (1968)  
5821       NaN  Goodare et al. (2003

### First glance: total casualty
In this dataset:

In [3]:
# how many people are tried in total
# df.sum(axis=0) Summing columns, or summing across rows, set axis=0
trials_clean['tried'].sum(axis=0)

43240

In [4]:
# how many people are killed in total
trials_clean['deaths'].sum(axis=0)

16333.0

So this is just a subset of the people tried and persecuted during the witch trial. According to [Wikipedia's Witch Trials in the early modern period](https://en.wikipedia.org/wiki/Witch_trials_in_the_early_modern_period), between 1400 to 1782, around 40,000 to 60,000 people were killed due to suspicion that they were practicing witchcraft. Some sources estimate that a total of 100,000 trials occurred at its maximum for a similar period.

## Q1: What time period (century/decade/year) witnessed the peak of witch trials?

### Witch Trials by century - line chart

In [5]:
# Group by century 
trials_by_century = trials_clean.groupby([pd.Grouper(key='century')]).count()[['tried']].reset_index()

# rename the columns
trials_by_century.columns = ['century', 'trials_count']
trials_by_century

century  trials_count
0     1300           150
1     1400           607
2     1500          2364
3     1600          7164
4     1700           652
5     1800             3

In [6]:
# line chart by altair
alt.Chart(trials_by_century).mark_line().encode(
    x=alt.X('century'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by century',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

![witch trials by century](./img/witch_trials_by_century.png)

#### The zenith between 1560 and 1630
The period between 1560 and 1630 was known as the 'Great Hunt', when 60% of the prosecution took place. Over the course of a century and a half, 80,000 people were tried for witchcraft and half of them were executed, often burned alive. This time period coincided with the Counter-Reformation and the European wars of religion. [Economists Peter T. Leeson and Jake Russ argued](https://www.peterleeson.com/witch_trials.pdf) that the surgence of European witch trials reflected non-price competition between the Catholic and Protestant churches for religious market share in confessionally contested parts of Christendom.

Simply put, as competing Catholic and Protestant churches vied to win over or retain their followers, they needed to make an impact – and witch trials were the battleground they chose ([the Guardian](https://www.theguardian.com/society/2018/jan/07/witchcraft-economics-reformation-catholic-protestant-market-share)).

### Witch Trials By Decade - line chart

In [7]:
# Group by decade 
trials_by_decade = trials_clean.groupby([pd.Grouper(key='decade')]).count()[['tried']].reset_index()

# rename the columns
trials_by_decade.columns = ['decade', 'trials_count']
trials_by_decade

decade  trials_count
0     1300            11
1     1310            21
2     1320            21
3     1330            25
4     1340             8
5     1350             4
6     1360             3
7     1370            15
8     1380            17
9     1390            25
10    1400            32
11    1410            20
12    1420            35
13    1430            52
14    1440            57
15    1450            90
16    1460            65
17    1470            68
18    1480           101
19    1490            87
20    1500            39
21    1510            43
22    1520            64
23    1530            73
24    1540           106
25    1550           111
26    1560           309
27    1570           291
28    1580           373
29    1590           955
30    1600           621
31    1610           878
32    1620           934
33    1630           724
34    1640          1111
35    1650           872
36    1660          1050
37    1670           419
38    1680           260
39    1690           295
40    1700           195
41    1710           107
42    1720           105
43    1730            68
44    1740            89
45    1750            62
46    1760             8
47    1770            11
48    1780             3
49    1790             4
50    1820             1
51    1830             1
52    1850             1

In [8]:
# line chart by altair
alt.Chart(trials_by_decade).mark_line().encode(
    x=alt.X('decade'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by decade',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

![witch trials by decade](./img/witch_trials_by_decade.png)

### Witch Trials By Year - line chart

In [9]:
# Group by year 
trials_by_year = trials_clean.groupby([pd.Grouper(key='year')]).count()[['tried']].reset_index()

# rename the columns
trials_by_year.columns = ['year', 'trials_count']

In [10]:
# line chart by altair
alt.Chart(trials_by_year).mark_line().encode(
    x=alt.X('year'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by year',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


alt.Chart(...)

![witch trials by year](./img/witch_trials_by_year.png)

According to the data we have, the 16 century witnessed the highest witch trials count. There are 1050 cases in 1660 alone. 